mne/cov.py

```diff
@@ -876,24 +938,41 @@ def _check_scalings_user(scalings):
 
 
 def _compute_covariance_auto(data, method, info, method_params, cv,
-                             scalings, n_jobs, stop_early, picks_list):
+                             scalings, n_jobs, stop_early, picks_list, rank):
     """Compute covariance auto mode."""
     # rescale to improve numerical stability
     _apply_scaling_array(data.T, picks_list=picks_list, scalings=scalings)
+    if rank != 'full':
```

Comment:
Should be a better test. If len(data) == 306 and rank == 306 this block should not be entered.

Reply:
A couple of things:

1. The check for this would actually be a bit annoying, as you need to then triage based on the various incantations of `rank` input, instead of just letting `_smart_eigh` be the one place this is handled / estimated.
2. In principle it should not matter. As long as we do the rotation properly, it should gracefully fall back to identical behavior.

I have now added a test that point (2) holds. It turns out that it did not before (whoops) but it does now!